<launch>
    <arg name="M_PI" value="3.14159265"/>

    
<!-- 13_Depth example -->
    <arg name="filepath" default="/home/mike/catkin_ws/src/StairwayDetection/examples/13_Depth.pcd"/>
    <arg name="origin" default="0 0 0 0 0 0"/>
    <arg name="fixed_frame_id" default="base_link"/>
    <arg name="child_frame_id" default="lidar"/>

<!-- stairs_descent_ECESAtrium_close -->
    <!-- <arg name="filepath" default="/home/mike/code_files/marble/pcds/stairs_descent_ECESAtrium_close.pcd"/>
    <arg name="origin" default="0 0 0 0 0 -2.4"/> -->

<!-- stairs_descent_ECESAtrium_far -->
    <!-- <arg name="filepath" default="/home/mike/code_files/marble/pcds/stairs_descent_ECESAtrium_far.pcd"/>
    <arg name="origin" default="0 0 0 0 0 -2.1"/> -->

<!--  -->
    <!-- <arg name="filepath" default="/home/mike/code_files/marble/pcds/stairs_descent_ECESAtrium_far.pcd"/>
    <arg name="origin" default="0 0 0 0 0 0"/> -->

<!-- STAIRS PUBLISHER -->
    <node pkg="stair_detection" type="example_stairs_publisher" name="stairs_publisher" output="screen"
    args="$(arg filepath) $(arg child_frame_id)"/>

    <include file="$(find curvy_terrain_mapper)/launch/curvy_terrain_mapper.launch">
        <arg name="fixed_frame_id" value="$(arg fixed_frame_id)"/>
    </include>

    <include file="$(find octomap_server)/launch/octomap_mapping.launch" if="$(arg enable_octomap)">
        <arg name="cloud_in"        value="/curvy_terrain_mapper/cost_cloud"/>
        <arg name="frame_id"        value="$(arg fixed_frame_id)"/>
        <arg name="base_frame_id"   value="$(arg child_frame_id)"/>
        <arg name="use_height_map"  value="true"/>
        <arg name="use_colored_map" value="false"/>
        <arg name="use_rough_map"   value="false"/>
        <arg name="filter_speckles" value="true"/>
        <arg name="filter_ground"   value="false"/>
        <arg name="resolution"      value="0.2"/>
        <arg name="edt/enable"      value="true"/>
    </include>

    
    <node pkg="tf2_ros" type="static_transform_publisher" name="TStair"
    args="$(arg origin) $(arg fixed_frame_id) $(arg child_frame_id)"/>

</launch>